plugin/pkg/scheduler/algorithm/priorities/util/non_zero.go

```diff
@@ -48,3 +53,68 @@ func GetNonzeroRequests(requests *api.ResourceList) (int64, int64) {
 	}
 	return outMilliCPU, outMemory
 }
+
+// FilterPodsByNameSpaces filters the pods based the given list of namespaces
+func FilterPodsByNameSpaces(names sets.String, pods []*api.Pod) []*api.Pod {
+	if len(pods) == 0 || len(names) == 0 {
+		return pods
+	}
+	result := []*api.Pod{}
+	for _, pod := range pods {
+		if names.Has(pod.Namespace) {
+			result = append(result, pod)
+		}
+	}
+	return result
+}
+
+// GetNamespacesFromPodAffinityTerm returns a set of names
+// according to the namespaces indicated in podAffinityTerm.
+// if the NameSpaces is nil considers the given pod's namespace
+// if the Namespaces is empty list then considers all the namespaces
+func GetNamespacesFromPodAffinityTerm(pod *api.Pod, podAffinityTerm api.PodAffinityTerm) sets.String {
+	names := sets.String{}
+	if podAffinityTerm.Namespaces == nil {
+		names.Insert(pod.Namespace)
+	} else if len(podAffinityTerm.Namespaces) != 0 {
+		for _, nameSpace := range podAffinityTerm.Namespaces {
+			names.Insert(nameSpace.Name)
+		}
+	}
+	return names
+}
+
+// NodeHasTopologyKey checks if nodeA and nodeB have same label value with given topologyKey as label key.
+// If the topologyKey is nil/empty, regard as the two nodes have same topologykey and label value.
+func NodesHaveSameTopologyKey(nodeA *api.Node, nodeB *api.Node, topologyKey string) bool {
+	if len(topologyKey) == 0 {
```

Comment:
Thanks. I would suggest to remove the part that says "by default they are unversioned.LabelHostname, unversioned.LabelZoneFailureDomain and unversioned.LabelZoneRegion)" because we may forget to update the comment if this set changes in the future; also, if someone looks for the --failure-domains command-line argument in the code, they should be able to find where the default value is set, so I think that part of the comment is not necessary.

BTW I would suggest adding a comment where those constants are defined (unversioned.LabelHostname etc.) saying to be sure to update the default for the --failure-domains flag in the scheduler if you change them. (I think one comment for the whole section is sufficient.)